### PR TITLE
OpenAI Compatible Server

### PR DESCRIPTION
**Pull request by e-cal**
Currently just wrapping stuff from `main` in a fastapi app.
Working towards full OpenAI compatible server.

### CODE REVIEW

**Comment by xjdr-alt** [comment]
This might be a good place to pull from https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/server.py

**Comment by e-cal** [comment]
First pass is done and fully working. Only implementing `/v1/chat/completions` (chat completions + streaming) right now. Pulled a little bit from sglang but mostly from scratch for simplicity.

Minimal working example:

```bash
python entropix/server.py
```

```python
from openai import OpenAI
from entropix.prompts import prompt

client = OpenAI(base_url="http://127.0.0.1:1337/v1")

completion = client.chat.completions.create(
  model="", # not used yet, model is bound to the server
  messages=[
    {"role": "user", "content": prompt}
  ],
)

print(completion.choices[0].message.content)
```

or streaming

```python
from openai import OpenAI
from entropix.prompts import prompt

client = OpenAI(base_url="http://127.0.0.1:1337/v1")

completion = client.chat.completions.create(
  model="", # not used yet, model is bound to the server
  messages=[
    {"role": "user", "content": prompt}
  ],
  stream=True,
)

for chunk in completion:
  content = chunk.choices[0].delta.content
  if content:
    print(content, end="", flush=True)
```

File: entropix/torch_weights.py
Comment (e-cal) [inline entropix/torch_weights.py:1]: Not related to this PR but fixed torch impl crashing for me. Can revert if you want.

**Comment by e-cal** [comment]
closed in favor of the `server` branch implementation

**Pull request closed without merging**